File: Android/iAthletics/src/com/appapro/iathletics/services/NextEventService.java
package com.appapro.iathletics.services;

import java.text.ParseException;
import java.text.SimpleDateFormat;

import com.appapro.iathletics.activities.iAthleticsActivity;
import com.appapro.iathletics.types.AppEvent;
import com.appapro.iathletics.types.iAthleticsImage;

public class NextEventService extends iAthleticsService
{	
	
	protected AppEvent nextEvent;
	
	public NextEventService(iAthleticsActivity activity)
	{
		super(activity);
	}
	
	public AppEvent getNextEvent()
	{
		this.requestService("GetNextEvent", null);
		
		return this.nextEvent;
	}

	protected void elementStart(String elementName)
	{
		// prepare to receive new event Info
		//
		if(elementName.compareTo("Event") == 0)
		{
			this.nextEvent = new AppEvent();
		}
	}

	protected void elementEnd(String elementName, String text)
	{
		// is this the event id
		//
		if(elementName.compareTo("eventId") == 0)
		{
			this.nextEvent.setEventId(Integer.valueOf(text));
		}
		
		// event date and time
		//
		else if(elementName.compareTo("EventDateTime") == 0)
		{
			SimpleDateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
			
			try
			{
				this.nextEvent.setEventDateTime(dateFormatter.parse(text));
			}
			catch (ParseException e)
			{
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
		}
		else if(elementName.compareTo("TeamName") == 0)
		{
			this.nextEvent.setTeamName(text);
		}
		
		else if(elementName.compareTo("SportName") == 0)
		{
			this.nextEvent.setSportName(text);
		}
		
		else if(elementName.compareTo("EventName") == 0)
		{
			this.nextEvent.setEventName(text);
		}
		else if(elementName.compareTo("IconUrl") == 0)
		{
			this.nextEvent.setImage(new iAthleticsImage(text, getActivity()));
		}
		else if(elementName.compareTo("AwayEvent") == 0)
		{
			this.nextEvent.setAwayEvent(Boolean.valueOf(text));
		}
		else if(elementName.compareTo("ConferenceEvent") == 0)
		{
			this.nextEvent.setConferenceEvent(Boolean.valueOf(text));
		}
		else if(elementName.compareTo("Notes") == 0)
		{
			this.nextEvent.setNote(text);
		}

		else if(elementName.compareTo("Results") == 0)
		{
			this.nextEvent.setResults(text);
		}
		
		// parse the venu information
		//
		else if(elementName.compareTo("Address1") == 0)
		{
			this.nextEvent.getVenu().setAddress1(text);
		}
		else if(elementName.compareTo("Address2") == 0)
		{
			this.nextEvent.getVenu().setAddress2(text);
		}	
		else if(elementName.compareTo("City") == 0)
		{
			this.nextEvent.getVenu().setCity(text);
		}
		else if(elementName.compareTo("State") == 0)
		{
			this.nextEvent.getVenu().setState(text);
		}
		else if(elementName.compareTo("Zip") == 0)
		{
			this.nextEvent.getVenu().setZip(text);
		}
		else if(elementName.compareTo("WebUrl") == 0)
		{
			this.nextEvent.getVenu().setWebUrl(text);
		}
		else if(elementName.compareTo("VenuName") == 0)
		{
			this.nextEvent.getVenu().setVenuName(text);
		}
	}
}

File: iPhone/Apex Cougars/Classes/NextEventService.m
//
//  NextEventService.m
//  iAthletics Framework
//
//  Created by Robert Signore on 6/15/10.
//  Copyright 2010 Appapro. All rights reserved.
//

#import "NextEventService.h"


@implementation NextEventService

@synthesize nextEvent;

-(void)dealloc
{
	[nextEvent release];
	
	[super dealloc];
}

-(void)requestService
{
	[self requestServiceName:@"GetNextEvent" parameters:nil];
}

#pragma mark -
#pragma mark iAthleticsService ABSTRACT methods


-(void) elementStart:(NSString *)elementName
{
	// prepare to receive new event Info
	//
	if([elementName compare:@"Event"] == NSOrderedSame)
	{
		AppEvent *newEvent = [[AppEvent alloc]init];
		self.nextEvent = newEvent;
		
		[newEvent release];
	}
}

-(void)elementEnd:(NSString *)elementName withText:(NSString *)text
{
	// is this the event id
	//
	if([elementName compare:@"eventId"] == NSOrderedSame)
	{
		self.nextEvent.eventId = [text intValue];
	}
	
	// event date and time
	//
	else if([elementName compare:@"EventDateTime"] == NSOrderedSame)
	{
		NSDateFormatter *dateFormatter = [[NSDateFormatter alloc]init];
		[dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss"];
		
		self.nextEvent.eventDateTime = [dateFormatter dateFromString:text];
		
		[dateFormatter release];
	}
	else if([elementName compare:@"TeamName"] == NSOrderedSame)
	{
		self.nextEvent.teamName = text;
	}
	
	else if([elementName compare:@"SportName"] == NSOrderedSame)
	{
		self.nextEvent.sportName = text;
	}
	
	else if([elementName compare:@"EventName"] == NSOrderedSame)
	{
		self.nextEvent.eventName = text;
	}
	
	else if([elementName compare:@"IconUrl"] == NSOrderedSame)
	{
		iAthleticsImage *newImage = [[iAthleticsImage alloc]initWithUrl:text];
		self.nextEvent.image = newImage;
		
		[newImage release];
	}
	
	else if([elementName compare:@"AwayEvent"] == NSOrderedSame)
	{
		self.nextEvent.awayEvent = [text boolValue];
	}
	
	else if([elementName compare:@"ConferenceEvent"] == NSOrderedSame)
	{
		self.nextEvent.conferenceEvent = [text boolValue];
	}

	else if([elementName compare:@"Notes"] == NSOrderedSame)
	{
		self.nextEvent.note = text;
	}

	else if([elementName compare:@"Results"] == NSOrderedSame)
	{
		self.nextEvent.results = text;
	}
	
	// parse the venu information
	//
	else if([elementName compare:@"Address1"] == NSOrderedSame)
	{
		self.nextEvent.venu.address1 = text;
	}
	else if([elementName compare:@"Address2"] == NSOrderedSame)
	{
		self.nextEvent.venu.address2 = text;
	}	
	else if([elementName compare:@"City"] == NSOrderedSame)
	{
		self.nextEvent.venu.city = text;
	}
	else if([elementName compare:@"State"] == NSOrderedSame)
	{
		self.nextEvent.venu.state = text;
	}
	else if([elementName compare:@"Zip"] == NSOrderedSame)
	{
		self.nextEvent.venu.zip = text;
	}
	else if([elementName compare:@"WebUrl"] == NSOrderedSame)
	{
		self.nextEvent.venu.webUrl = text;
	}
	else if([elementName compare:@"VenuName"] == NSOrderedSame)
	{
		self.nextEvent.venu.venuName = text;
	}
    
    // added for new result view
    //
    else if([elementName compare:@"ResultId"] == NSOrderedSame)
    {
        // this tag will come in in the form of </ ResultId xlns:nil> if null
        // text will be an empty string
        //
        if(text != nil)
        {
            self.nextEvent.resultId = [text intValue];
        }
    }
    else if([elementName compare:@"ResultDetailsUrl"] == NSOrderedSame)
    {
        self.nextEvent.resultDetailsUrl = text;
    }
}


@end
